packages/utils/src/deepClone.ts

```diff
@@ -0,0 +1,15 @@
+export const deepClone = <T>(value: T): T => {
```

Comment:
I would suggest little bit restrict values that can be passed, because you can pass anything now I think, but I am not sure how JSON will behave if you will pass object with BigInt or other special types. It will work for structuredClone but JSON could mess it up, so we should limit it only types that will work 100% for both solutions.

Reply:
And now we are back to square 1 from which the conversation has started hehe :D Yeah, makes sense.

Reply:
I agree that we should consider that objects are going to be stringified when using `JSON.pasrse(JSON.stringify...`.
I am not worried about current places where it is used but in potential new uses where we think that this behaves like  `structuredClone` and in in some scenarios it will not.

```javascript
const appointment = {
    location: 'Prague',
    day: new Date(),
}

const whenStructuredClone = structuredClone(appointment)
whenStructuredClone.day.getDate()
// result -> 25

const whenStringified = JSON.parse(JSON.stringify(appointment))
whenStringified.day.getDate()
// result -> Uncaught TypeError: whenStringified.day.getDate is not a function
```

And the same for nested objects. And the same behavior can be expected from other objects like `BigInt`.

Restricting the use to types that will work 100% for both could be a solution but if I am not mistaken I think it should iterate over the whole object and nested potential objects to make sure there is no nested `BigInt` or other not working type.

Reply:
I think type like this should do the job:

```tsx
import { Primitive } from 'type-fest'

type PrimiteRecord = Record<string, Primitive | PrimitiveRecord>
type PrimitiveArray = Array<Primitive | PrimitiveRecord | PrimitiveArray>
```